packages/scripts/utils/config.js

```diff
@@ -206,17 +206,15 @@ function getWebpackEntryPoints() {
 
 	// 2. Checks whether any block metadata files can be detected in the defined source directory.
 	//    It scans all discovered files looking for JavaScript assets and converts them to entry points.
+	const srcDirectory = fromProjectRoot( getWordPressSrcDirectory() ) + sep;
 	const blockMetadataFiles = glob(
```

Comment:
Is there an issue with the `glob`? If that's the case, we should update all occurrences of `glob` in this file. Instead of changing the pattern, it can also work with `cwd` option:

```js
glob(
    '**/block.json',
    {
         absolute: true,
         cwd: fromProjectRoot( getWordPressSrcDirectory() ),
    }
);
```

I haven't tested it, but at the moment in [fast-glob](https://www.npmjs.com/package/fast-glob) `cwd` defaults to `process.cwd()` so that might be the equivalent.

Reply:
Makes sense! It is a simpler change, and I confirmed that it works correctly on Windows OS and WSL2 (Ubuntu) 👍

Reply:
There are two more places with `glob` which might be impacted by the same issue:

https://github.com/WordPress/gutenberg/blob/e9a946a38d2f42d5ad1fa7e1edfbf1cb6a7bbb84/packages/scripts/utils/config.js#L260

https://github.com/WordPress/gutenberg/blob/e9a946a38d2f42d5ad1fa7e1edfbf1cb6a7bbb84/packages/scripts/utils/config.js#L304

I really appreciate your contribution to make tooling work correctly with all operating systems.

Reply:
Updated 👍

- ✅ Tested on Windows 11 and WSL2 (Ubuntu)
- ✅Additionally confirmed that it builds correctly even when only `index.js` exists in the `src` directory.

Reply:
Awesome, it's a pleasure to collaborate with you :)